.rubocop.yml

```diff
@@ -1,6 +1,199 @@
-inherit_from: .rubocop_todo.yml
+
 AllCops:
   Include:
     - Rakefile
     - Gemfile
     - '*.gemspec'
+
+# I like this Metric in principle, but I don't like the default max of
+# 15.
+#
+# Also, as per Metrics/ClassLength IMO this kind of limit should not
+# apply to test code (I get up to 318 over there).
+#
+Metrics/AbcSize:
+  Max: 100
+  Exclude:
+    - 'test/**/*.rb'
+
+# Broadly speaking, test code gets a pass for most of the Metrics family.
+#
+# IMO test code is not the place get pedantic about class length,
+# method complexity, etc.  One should be encouraged to add more tests
+# with minimal friction, not forced to make a hard choice between
+# cutting tests or splitting up my test suites.
+#
+Metrics/BlockLength:
+  Exclude:
+    - 'test/**/*.rb'
+
+# Ditto I believe in more permissive rules in test code.
+#
+Metrics/ClassLength:
+  Max: 400
+  Exclude:
+    - 'test/**/*.rb'
+
+# Redis::ScriptManager#eval_gently is indeed overly complex but I am
+# happy with it until the next time something goes wrong.
+#
+# Also ditto I believe in more permissive rules in test code.
+#
+Metrics/CyclomaticComplexity:
+  Max: 30
+  Exclude:
+    - 'test/**/*.rb'
+Metrics/MethodLength:
+  Max: 100
+  Exclude:
+    - 'test/**/*.rb'
+Metrics/PerceivedComplexity:
+  Max: 21
+  Exclude:
+    - 'test/**/*.rb'
+
+# I put extra spaces in a lot of expressions for a lot of different
+# reasons, including especially readability.
+#
+# I reject these cops.
+#
+Layout:
+  Enabled: false
+
+# I like a lot of the Lint tests, but not these.
+#
+Lint/AmbiguousBlockAssociation:           # obnoxiously rejects idiomatic Ruby
+  Enabled: false
+
+# This does no more than insist I type "format" instead of "sprintf",
+# where the two are aliases.
+#
+Style/FormatString:
+  Enabled: false
+
+# This complains that:
+#
+#   if foo
+#     foo.bar
+#   end
+#
+# could instead be:
+#
+#   foo.bar if foo
+#
+# ...which I reject, not least because the former is a lot easier to
+# fit into an 80 character line.
+#
```

Comment:
`GuardClause` should defer to `LineLength` ... i.e. it won't ask you for a guard clause if that would take the line over the length limit.

Not objecting to this change, just sharing some relevant information.